### PR TITLE
Test whether copy of directory will copy the mtime.

### PR DESCRIPTION
**Pull request by kersommoura**
* Test:

  1. Create and sync a repository.
  2. Publish the repository with the option `generate_sqlite` as True.
  3. Verify the `mtime` for the sqlite files.
  4. Upload a RPM package to the repository.
  5. Sync the repository again.
  6. Assert that `mtime` for old sqlite files remain the same.

### CODE REVIEW

File: pulp_smash/tests/rpm/api_v2/test_copy_mtime.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        * `Pulp Smash #720 <https://github.com/PulpQE/pulp-smash/issues/720>`_
+        * `Pulp #2783 <https://pulp.plan.io/issues/2783>`_
+
+        Do the following:
+
+        1. Create and sync a repository.
+        2. Publish the repository with the option ``generate_sqlite``
+           as True.
+        3. Verify the ``mtime`` for the sqlite files.
+        4. Upload a RPM package to the repository.
```
Comment (omaciel) [inline pulp_smash/tests/rpm/api_v2/test_copy_mtime.py]: Is this a new package or an existing package already synced?

File: pulp_smash/tests/rpm/api_v2/test_copy_mtime.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        5. Sync the repository again.
+        6. Assert that ``mtime`` for old sqlite files remain the same.
```
Comment (omaciel) [inline pulp_smash/tests/rpm/api_v2/test_copy_mtime.py]: Is it necessary to re-publish? Sorry to ask, but I'm more familiar with Satellite terms and flow :)

File: pulp_smash/tests/rpm/api_v2/test_copy_mtime.py (hunk 1 above)
Comment (kersommoura) [reply pulp_smash/tests/rpm/api_v2/test_copy_mtime.py]: It is a different package. I used as feed url a unsigned package, and to upload a signed package.

File: pulp_smash/tests/rpm/api_v2/test_copy_mtime.py (hunk 2 above)
Comment (kersommoura) [reply pulp_smash/tests/rpm/api_v2/test_copy_mtime.py]: It was not required as part of steps to reproduce the issue. 
Sync in pulp will download the contents of the remote repository and store in our new repository. It is connected to the importer.
Publish will make available for the end user this new content. It is related to the distributor.
I will test if there is any difference, re-publishing before making the assertions.

File: pulp_smash/tests/rpm/api_v2/test_copy_mtime.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            'id': repo['distributors'][0]['id'],
+            'override_config': {'generate_sqlite': True},
+        })
+        repo = client.get(repo['_href'], params={'details': True})
+        path = os.path.join(
+            _PATH,
+            repo['distributors'][0]['config']['relative_url'],
+            'repodata'
+        )
+        system = cfg.get_systems('pulp cli')[0]
```
Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_copy_mtime.py]: FYI, the `pulp cli` role indicates that Pulp's CLI tools are installed. In other words, the role indicates that `pulp-admin` is available. It doesn't indicate that shell access is available. The `shell` role indicates that shell access is available.

The host that fulfills the `api` role runs Apache, and it's therefore likely to also have access to the sqlite files relevant to this test.

File: pulp_smash/tests/rpm/api_v2/test_copy_mtime.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        sudo = () if utils.is_root(cfg, system) else ('sudo',)
+        cli_client = cli.Client(cfg, cli.echo_handler)
+        responses = cli_client.run(sudo + (
+            'ls', '-l', '--full-time', path
```
Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_copy_mtime.py]: `ls` produces human-readable output. It's output isn't consistent over time or over different distributions, and therefore, it's output shouldn't be parsed by programming tools. Something like `stat --printf %Y /path/to/file` is more appropriate.

File: pulp_smash/tests/rpm/api_v2/test_copy_mtime.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        body = gen_repo()
+        body['importer_config']['feed'] = RPM_UNSIGNED_FEED_URL
+        body['distributors'] = [gen_distributor()]
+        repo = client.post(REPOSITORY_PATH, body)
+        self.addCleanup(client.delete, repo['_href'])
+        repo = client.get(repo['_href'], params={'details': True})
+        utils.sync_repo(cfg, repo)
+        utils.publish_repo(cfg, repo, {
+            'id': repo['distributors'][0]['id'],
+            'override_config': {'generate_sqlite': True},
```
Comment (Ichimonji10) [inline pulp_smash/tests/rpm/api_v2/test_copy_mtime.py]: If `generate_sqlite` can be set when the distributor is created, then that would be a more clean and concise way of doing things.

**Comment by Ichimonji10** [comment]
Rebased and merged as 29cf01b7489ca4c96bfdd9d6c641a578df2a933f, with a follow-up commit of 8157631db1ffbf86d0c10530198b69f418c8a018.

**Pull request closed without merging**